src/libs/SubscriptionUtils.ts

```diff
@@ -1,9 +1,121 @@
+
+const PAYMENT_STATUSES = {
+    POLICY_OWNER_WITH_AMOUNT_OWED: 'policy_owner_with_amount_owed',
+    POLICY_OWNER_WITH_AMOUNT_OWED_OVERDUE: 'policy_owner_with_amount_owed_overdue',
+    OWNER_OF_POLICY_UNDER_INVOICING: 'owner_of_policy_under_invoicing',
+    OWNER_OF_POLICY_UNDER_INVOICING_OVERDUE: 'owner_of_policy_under_invoicing_overdue',
+    BILLING_DISPUTE_PENDING: 'billing_dispute_pending',
+    CARD_AUTHENTICATION_REQUIRED: 'authentication_required',
+    INSUFFICIENT_FUNDS: 'insufficient_funds',
+    CARD_EXPIRED: 'expired_card',
+    CARD_EXPIRE_SOON: 'card_expire_soon',
+    RETRY_BILLING_SUCCESS: 'retry_billing_success',
+    RETRY_BILLING_ERROR: 'retry_billing_error',
+    GENERIC_API_ERROR: 'generic_api_error',
+};
+
+let amountOwed: OnyxValues[typeof ONYXKEYS.NVP_PRIVATE_AMOUNT_OWED];
+Onyx.connect({
+    key: ONYXKEYS.NVP_PRIVATE_AMOUNT_OWED,
+    callback: (value) => {
+        if (!value) {
+            return;
+        }
+
+        amountOwed = value;
+    },
+});
+
+let stripeCustomerId: OnyxValues[typeof ONYXKEYS.NVP_PRIVATE_STRIPE_CUSTOMER_ID];
+Onyx.connect({
+    key: ONYXKEYS.NVP_PRIVATE_STRIPE_CUSTOMER_ID,
+    callback: (value) => {
+        if (!value) {
+            return;
+        }
+
+        stripeCustomerId = value;
+    },
+});
+
+let billingDisputePending: OnyxValues[typeof ONYXKEYS.NVP_PRIVATE_BILLING_DISPUTE_PENDING];
+Onyx.connect({
+    key: ONYXKEYS.NVP_PRIVATE_BILLING_DISPUTE_PENDING,
+    callback: (value) => {
+        if (!value) {
+            return;
+        }
+
+        billingDisputePending = value;
+    },
+});
+
+let billingStatus: OnyxValues[typeof ONYXKEYS.NVP_PRIVATE_BILLING_STATUS];
+Onyx.connect({
+    key: ONYXKEYS.NVP_PRIVATE_BILLING_STATUS,
+    callback: (value) => {
+        if (!value) {
+            return;
+        }
+
+        billingStatus = value;
+    },
+});
+
+let billingGracePeriod: OnyxValues[typeof ONYXKEYS.NVP_PRIVATE_OWNER_BILLING_GRACE_PERIOD_END];
+Onyx.connect({
+    key: ONYXKEYS.NVP_PRIVATE_OWNER_BILLING_GRACE_PERIOD_END,
+    callback: (value) => {
+        if (value === undefined) {
+            return;
+        }
+
+        billingGracePeriod = value;
+    },
+});
+
+let fundList: OnyxValues[typeof ONYXKEYS.FUND_LIST];
+Onyx.connect({
+    key: ONYXKEYS.FUND_LIST,
+    callback: (value) => {
+        if (!value) {
+            return;
+        }
+
+        fundList = value;
+    },
+});
+
+let billingStatusFailed: OnyxValues[typeof ONYXKEYS.SUBSCRIPTION_RETRY_BILLING_STATUS_FAILED];
+Onyx.connect({
+    key: ONYXKEYS.SUBSCRIPTION_RETRY_BILLING_STATUS_FAILED,
+    callback: (value) => {
+        if (value === undefined) {
+            return;
+        }
+
+        billingStatusFailed = value;
+    },
+});
+
+let billingStatusSuccessful: OnyxValues[typeof ONYXKEYS.SUBSCRIPTION_RETRY_BILLING_STATUS_SUCCESSFUL];
+Onyx.connect({
+    key: ONYXKEYS.SUBSCRIPTION_RETRY_BILLING_STATUS_SUCCESSFUL,
+    callback: (value) => {
```

Comment:
I think this should be initialized with extra option ` initWithStoredValues: false` according to the [doc](https://docs.google.com/document/d/1jXV_vIsOGZuL0dMiuXDWRETAFRX64ODVAhWPtiuFXT8/edit?pli=1#bookmark=id.ha6pkejex3ep)